### ISSUE TITLE
how to cite

### ISSUE THREAD

**Issue by tpapp** [robertfeldt/BlackBoxOptim.jl#76]
This package helped me a lot with an MCMC problem (finding an initial point for starting the tuning, with a heavily multimodal likelihood), and I would like to mention it, or papers it is related to, in the paper the results ended up in. A reference, or ideally a BibTeX entry, would be helpful.

*(sorry if this is mentioned somewhere, could not find it)*

**Comment by robertfeldt** [comment]
Thanks, I'm very glad to hear that. For now maybe use something like this:

@misc{Feldt2018,
  author = {Robert Feldt},
  title = {BlackBoxOptim.jl},
  year = {2018},
  publisher = {GitHub},
  journal = {GitHub repository},
  howpublished = {\url{https://github.com/robertfeldt/BlackBoxOptim.jl}},
  commit = {ADDSAHOFCOMMITYOUUSEDHERE}
}

Since @alyst has contributed quite a lot you might want to use instead:

  author = {Robert Feldt and Alexey Stukalov}

but formally, LICENSE-wise, I guess I'm still the author. @alyst if you have any input on this feel free to share; I want to do this right.

Longer-term I plan to publish a small arxiv.org report on this; then you can cite that instead.

If you report/paper is published please share so we can add to the repo as a list of impacts.

**Comment by robertfeldt** [comment]
Should have been "ADDSHAOFCOMMITYOUUSEDHERE"... ;)

**Comment by tpapp** [comment]
Thanks, I think I will go with the tagged version.

When the report on arxiv comes out, please consider mentioning it in this issue (which you might want to keep open, so that people interested in the same information could subscribe).

**Comment by robertfeldt** [comment]
Ok, great, will do.

**Comment by alyst** [comment]
Of course I would be very grateful, if I'm mentioned, but it's also totally fine to do it according to the license.

@tpapp Out of curiosity, which method did you use?

**Comment by tpapp** [comment]
@alyst: 

- Betancourt (2017), implemented in [DynamicHMC.jl](https://github.com/tpapp/DynamicHMC.jl) for the HMC;
- Gallant & McCulloch (2009) for the indirect likelihood, as implemented [here (warning: WIP)](https://github.com/tpapp/IndirectLikelihood.jl)

**Comment by alyst** [comment]
@tpapp Thanks! I missed that there's `DynamicHMC.jl`, definitely will have a look. Which BlackBox optimization method did you use to find the initial point?

**Comment by tpapp** [comment]
@alyst: The default, which I think is `adaptive_de_rand_1_bin_radiuslimited()`. I have to admit that I don't know much about these methods, but they seem very robust. I just need a "good enough" starting point, not a global optimum, but I checked some cases and it seems to find the global one. Dimensions are around 10-30 ATM, will grow later.

**Comment by robertfeldt** [comment]
Yes, the default is very robust and should scale better than many alternatives. The NES family (dxnes probably the best example in it) can be better for low dimensions but tend to have problems to scale up to more dimensions IMHO.

**Comment by robertfeldt** [comment]
@alyst I suggest we write a short arxiv report with both of us as authors and keep the license as is, if that is ok to you. Since when I use this for consultancy work etc I would rather not have to take license concerns into account. I'm sure they are also solvable but here I'm mostly after sharing the credit (based on GitHub stats it seems the commits are about 76% me and 23% you and 1% other at the moment). Hope ok.

When it comes to the report, I have my hands full the coming month but would then be interested in doing something. Maybe we could even collect a few more "hard" problems (@tpapp can you contribute one of yours?) and do some new performance screening to include the guide for selecting an alg in the report?

**Comment by tpapp** [comment]
@robertfeldt: sure, the current data I am working with is confidential, but I will think of something. I am very busy now with this project, but expect to have time in February. Please let me know when you get started and I will contribute.

**Comment by alyst** [comment]
@robertfeldt That would be great! Pls let me know when it will come to planning the report. The problems I was applying BBO to are sufficiently "hard" (~10-1000 dims, 2 objectives). Unfortunately, they are also too complicated/specific to really consider for the short report.
